src/server.js

```diff
@@ -168,9 +168,41 @@ server
             </script>`
             : ''
         }
+        ${
+          runtimeConfig.MATOMO_URL && runtimeConfig.MATOMO_SITE_ID
+            ? `
+            <!-- Matomo -->
+            <script type="text/javascript">
+                var _paq = window._paq || [];
+                /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+                _paq.push(['trackPageView']);
+                _paq.push(['enableLinkTracking']);
+                (function () {
+                    var u = "${runtimeConfig.MATOMO_URL}";
+                    _paq.push(['setTrackerUrl', u + 'matomo.php']);
+                    _paq.push(['setSiteId', '${runtimeConfig.MATOMO_SITE_ID}']);
+                    var d = document, g = d.createElement('script'), s = d.getElementsByTagName('script')[0];
+                    g.type = 'text/javascript';
+                    g.async = true;
+                    g.defer = true;
+                    g.src = u + 'matomo.js';
+                    s.parentNode.insertBefore(g, s);
+                })();
+            </script>
+            <!-- Matomo End -->`
+            : ''
+        }
 
     </head>
     <body>
+        ${
+          runtimeConfig.MATOMO_URL && runtimeConfig.MATOMO_SITE_ID
+            ? `
+            <!-- Matomo Image Tracker-->
+            <img referrerpolicy="no-referrer-when-downgrade" src="${runtimeConfig.MATOMO_URL}matomo.php?idsite=${runtimeConfig.MATOMO_SITE_ID}&amp;rec=1" style="border:0" alt="" />
```

Comment:
one small thing here!  I think this would be 

`<img referrerpolicy="no-referrer-when-downgrade" src="${runtimeConfig.MATOMO_URL}/matomo.php?idsite=${runtimeConfig.MATOMO_SITE_ID}&amp;rec=1" style="border:0" alt="" />`

(missing `/`)  Otherwise your url would have to end in a trailing slash, which might be fine but also odd that you would have to know to end your site in a slash.  What do you think?

after this, looks good!

Reply:
The reason I did this is because the slash is there as well in the other part of the config where it just asks for the url. I didn't test if the slash was necessary there. Second thing I was thinking is that they will probably check out their config with this same code cause don't know how else you can get the site ID. I was thinking if they copied the URL from there (which has trailing slash) it would then create errors.

I also thought the same like you that it's pretty confusing. I guess it's confusing depending from which point you view it and could be either way confusing. 

I'll add another commit to what you said. Maybe my logic is too far fetched.

Reply:
I see what you mean but I think this is cleaner for the person using the URL as a variable, keep the ugly stuff in code ;)  Thank you!